tests/Functional/Fixtures/Symfony/AppCache.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace FOS\HttpCache\Tests\Functional\Fixtures\Symfony;
+
+use FOS\HttpCache\SymfonyCache\EventDispatchingHttpCache;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+
+class AppCache extends EventDispatchingHttpCache
+{
+    public function handle(Request $request, $type = HttpKernelInterface::MASTER_REQUEST, $catch = true)
+    {
+        $response = parent::handle($request, $type, $catch);
+
+        if ($response->headers->has('X-Symfony-Cache')) {
+            if (false !== strpos($response->headers->get('X-Symfony-Cache'), 'miss')) {
+                $state = 'MISS';
+            } elseif (false !== strpos($response->headers->get('X-Symfony-Cache'), 'fresh')) {
+                $state = 'HIT';
+            } else {
+                $state = 'UNDETERMINED';
```

Comment:
well, the response could miss that header, or it could contain neither miss nor fresh. i preferred this over defaulting to MISS or HIT in that case so that we see that there is a problem.